fix: Correct the return type of fetch_channels

retr0-init opened the pull request:
## Pull Request Type
<!-- Check the appropriate option -->

- [ ] Feature addition
- [x] Bugfix
- [ ] Documentation update
- [ ] Code refactor
- [ ] Tests improvement
- [ ] CI/CD pipeline enhancement
- [ ] Other: [Replace with a description]

## Description
<!-- Provide a clear and concise description of the purpose of this PR and why it should be merged -->
<!-- If your code adds or changes features, a usage example would be helpful -->
Correct the return type of `guild.fetch_channels` from VoiceChannel to `TYPE_GUILD_CHANNEL`

## Changes
<!-- List the changes you have made in a bullet-point format -->
- Change the return type of the function `fetch_channels` from `TYPE_VOICE_CHANNEL` to `TYPE_GUILD_CHANNEL`

## Related Issues
<!-- If this PR is related to any open issues, please mention them using "#ISSUENUMBER" -->


## Test Scenarios
<!-- Provide clear instructions on how to test your changes, where applicable - if no tests are required, explain why -->


## Python Compatibility
<!-- Testing 3.11 is not strictly required, but it would be nice if you could confirm that it works. -->
- [x] I've ensured my code works on Python `3.10.x`
- [x] I've ensured my code works on Python `3.11.x`


## Checklist
<!-- If you have not completed all of the following, there is a good chance your PR will not be merged. -->
- [x] I've run the `pre-commit` code linter over all edited files
- [x] I've tested my changes on supported Python versions
- [x] I've added tests for my code, if applicable
- [x] I've updated / added  documentation, where applicable

LordOfPolls committed:
5.11.0

5.11.0

retr0-init committed:
fix: Correct the return type of fetch_channels

Signed-off-by: retr0-init <146389702+retr0-init@users.noreply.github.com>

AstreaTSS commented on interactions/models/discord/guild.py:
```diff
@@ -2263,7 +2263,7 @@ def get_channels(self) -> List["models.TYPE_VOICE_CHANNEL"]:
         return [self._client.get_channel(channel_id) for channel_id in self._channel_ids]
 
-    async def fetch_channels(self, *, force: bool = False) -> List["models.TYPE_VOICE_CHANNEL"]:
+    async def fetch_channels(self, *, force: bool = False) -> List["models.TYPE_GUILD_CHANNEL"]:
```
This is meant to be `TYPE_VOICE_CHANNEL` due to it being a `GuildWidget`, so this should be reverted.

retr0-init replied:
This change is reverted

AstreaTSS reviewed (changes requested):
free pr (though with a change needed)

retr0-init committed:
fix: Revert return type of fetch_channels for GuildWidget

This is meant to be TYPE_VOICE_CHANNEL due to it being a GuildWidget, so this should be reverted.
https://github.com/interactions-py/interactions.py/pull/1622#pullrequestreview-1902234794

Signed-off-by: retr0-init <146389702+retr0-init@users.noreply.github.com>
